blockchain/blockchain.go

```diff
@@ -187,8 +186,6 @@ type BlockChain struct {
 	validator  Validator  // block and state validator interface
 	vmConfig   vm.Config
 
-	badBlocks *lru.Cache // Bad block cache
```

Comment:
Can I ask why you change the bad block storing method from cache to db?

Reply:
Ethereum also stores badblocks into DB now.
Plus if the memory runs out when a node dies, badblocks cannot be retrieved, which makes it impossible to debug badblocks when network goes down.

Reply:
@hqjang-pepper Oh, I see. Is there any related ethereum PR? It would be helpful to understand the history.

Reply:
Checkout this PR : https://github.com/ethereum/go-ethereum/pull/21827